diff --git a/src/DrupalSystem/DrupalSystem.php b/src/DrupalSystem/DrupalSystem.php
index 4bc50ce..0a856c3 100644
--- a/src/DrupalSystem/DrupalSystem.php
+++ b/src/DrupalSystem/DrupalSystem.php
@@ -53,9 +53,22 @@ class DrupalSystem implements DrupalSystemInterface {
    * {@inheritdoc}
    */
   function getActiveExtensions() {
-    // Doing this directly tends to be a lot faster than system_list().
-    return db_query("SELECT name, type from {system} WHERE status = 1")
-      ->fetchAllKeyed();
+    try {
+      // Doing this directly tends to be a lot faster than system_list().
+      return db_query("SELECT name, type from {system} WHERE status = 1")
+        ->fetchAllKeyed();
+    }
+    catch (\DatabaseConnectionNotDefinedException $e) {
+      // During install, the database is not available.
+      // At this time only the system module is "installed".
+      return array('system' => 'module');
+    }
+    catch (\PDOException $e) {
+      // Some time later during install, there is a database but not yet a system table.
+      // At this time only the system module is "installed".
+      // @todo Check if this is really a "Table 'system' doesn't exist'" exception.
+      return array('system' => 'module');
+    }
   }
 
   /**
