diff --git a/nlmfield.module b/nlmfield.module
index 330e463..69a2c8a 100644
--- a/nlmfield.module
+++ b/nlmfield.module
@@ -134,6 +134,7 @@ function nlmfield_ctools_plugin_api($module, $api) {
  * dependant columns. If XML is empty, we compute it from all other values.
  */
 function nlmfield_compute_values(&$values, $field_type, $namespace = 'nlm') {
+  nlmfield_load_library();
   $dom = new BetterDOMDocument($values['xml'], $namespace);
 
   foreach (nlmfield_column_bits('nlmfield_contributor') as $column => $bit) {
